src/include/btree_inline.h

```diff
@@ -2158,7 +2158,7 @@ __wt_split_descent_race(WT_SESSION_IMPL *session, WT_REF *ref, WT_PAGE_INDEX *sa
      * this code, don't re-order that acquisition with this check.
      */
     WT_BARRIER();
-    WT_INTL_INDEX_GET(session, ref->home, pindex);
+    pindex = WT_INTL_INDEX_GET_SAFE(ref->home);
```

Comment:
Could you please give the reason for the change in the macro? It seems like you wanted to exclude the following assertion.
`  WT_ASSERT(session, __wt_session_gen(session, WT_GEN_SPLIT) != 0);`

Reply:
for this call
```
if (__wt_split_descent_race(cbt->ref, pindex))
            skiplow = skiphigh = 0;
```
we have descended to the leaf page and has exited the split gen.

Reply:
However, seem like this function is also called at the internal page level. Wouldn't the assertion be significant there?

Reply:
We are just trying to get the new page index. We don't care if it is locked for split or not. So I think it should be ok.

Reply:
We check the race when we have descended to the child page. At this stage, we have lost our hazard pointer to the internal page. Therefore, it is no longer "SAFE".

Reply:
I think the naming causes confusion here. It is only OK to call `WT_INTL_INDEX_GET_SAFE` when you know the memory couldn't possibly be freed by anything else (for example, if you are doing a split and know nothing else can be, or have exclusive access to the tree).
All other users need to call the non-safe version - if they don't then a split can cause unexpected behavior.

Regardless - I'm pretty sure that all callers of split_descent_race should meet the requirements. If they had exclusive access, they shouldn't need to check for structural changes.

Reply:
yes, I agree.